app/helpers/document_helper.rb

```diff
@@ -24,4 +24,12 @@ def created_by(document)
   def document_name_and_reference(document)
     "#{document.name}#{document.numbers? ? " - #{document.numbers}" : ''}"
   end
+
+  def titled_reference_or_file_name(document)
+    document.numbers? ? "Reference: #{document.numbers}" : "File name: #{document.name}"
+  end
+
+  def reference_or_file_name(document)
+    document.numbers.presence || document.name
+  end
```

Comment:
I guess we currently have to use helper methods given that the logic is needed in both the views and the dreaded `PlanningApplicationPresenter`.

Reply:
I did a Component treatment which you've seen - I think comment was overkill.
I didn't want to just add it without tests or rather I would have had to have tested both cases with a system test which is slow and confusing.
I'm definitely open to ideas or it just becomes 2023 issue.